### ISSUE TITLE
Keywords button vanishes when entering incorrect phrase.

### ISSUE THREAD

**Issue by GwydionJon** [ssciwr/adaptivefiltering#274]
![image](https://user-images.githubusercontent.com/35529247/156397708-530dd076-ec6e-40c0-9da4-3663f626ad1f.png)

Not sure if this is just a visual bug on my machine or not.
It comes back after clicking around it.

**Comment by dokempf** [comment]
I can reproduce this but I think it is intended behaviour of the `TagsInput` widget.

**Comment by zaw-shinoto** [comment]
I can reproduce something which does not seem to be a bug. First of all, from the bug description I do not see where the bug happened but assume it is in `pipeline_tuning(dataset)`.

The first visualization (click on Preview) without keyword produces tab \#0, the second, where I entered `mariatest` retains this keyword, and the seventh, where I added `test` shows both. Screenshots added. 

![image](https://user-images.githubusercontent.com/25264488/156477541-0a4bab33-94d8-472f-b628-31d58aa6e870.png)

<img width="927" alt="Bildschirmfoto 2022-03-03 um 10 13 46" src="https://user-images.githubusercontent.com/25264488/156477601-de88c2a9-d069-409c-8d6b-9c6807852a68.png">

<img width="932" alt="Bildschirmfoto 2022-03-03 um 10 13 58" src="https://user-images.githubusercontent.com/25264488/156477613-076ec55c-0e11-48c7-b0f1-752b0e36819e.png">
